odoo/addons/base/tests/test_res_lang.py

```diff
@@ -40,3 +51,41 @@ def test_00_intersperse(self):
         assert intersperse("abc12", [3], '.') == ('abc12', 0)
         assert intersperse("abc12", [2], '.') == ('abc12', 0)
         assert intersperse("abc12", [1], '.') == ('abc1.2', 1)
+
+    def test_res_lang_decimal_point_preview(self):
+        self.assertEqual(self.lang_0.decimal_point_preview, '99.00')
+        self.lang_0.write({'decimal_point': ','})
+        self.assertEqual(self.lang_0.decimal_point_preview, '99,00')
+
+    def test_res_lang_thousand_sep_preview(self):
+        self.assertEqual(self.lang_0.thousand_sep_preview, '999,999,999')
+        self.lang_0.write({'thousands_sep': '.'})
+        self.assertEqual(self.lang_0.thousand_sep_preview, '999.999.999')
+
+    @freeze_time("2020-01-01 16:00:00")
+    def test_res_lang_date_format_preview(self):
+        lang_0, lang_1 = self.env['res.lang'].with_context(active_test=False).search(
+            [('iso_code', 'in', ['en', 'ar'])]
+        )
+
+        self.assertEqual(lang_0.date_format_preview, '01/01/2020')
+        self.assertEqual(lang_1.date_format_preview, '01 يناير, 2020')
```

Comment:
Use any language you want, provided it is based on latin charset. If in the future we need to change this test we (at lease I) can't even type in arabic, let alone verify what you wrote.
Could you change the language in this test ?